src/transformStateWithClones.js

```diff
@@ -8,6 +8,45 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const tempState = {};
+  const copyState = [];
+
+  Object.assign(tempState, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(tempState, action.extraData);
+        copyState.push(Object.assign({}, tempState));
```

Comment:
```suggestion

```

Reply:
You can create an additional variable before the switch case, then go through the switch, and after it make a push to avoid repeating code